src/Workspaces/CoreTest/Execution/Extensions.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Execution;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.UnitTests.Execution
+{
+    internal static class Extensions
+    {
+        public static ChecksumObjects<ProjectChecksumObject> ToProjectObjects(this ChecksumCollection collection, ISolutionChecksumService service)
+        {
+            Contract.ThrowIfFalse(collection.Kind == WellKnownChecksumObjects.Projects);
+            return new ChecksumObjects<ProjectChecksumObject>(service, collection);
+        }
+
+        public static ChecksumObjects<DocumentChecksumObject> ToDocumentObjects(this ChecksumCollection collection, ISolutionChecksumService service)
+        {
+            Contract.ThrowIfFalse(collection.Kind == WellKnownChecksumObjects.Documents || collection.Kind == WellKnownChecksumObjects.TextDocuments);
+            return new ChecksumObjects<DocumentChecksumObject>(service, collection);
+        }
+    }
+
+    internal class ChecksumObjects<T> : ChecksumObject where T : ChecksumObject
+    {
+        private readonly ChecksumCollection _collection;
```

Comment:
this is also fixed here.
https://github.com/dotnet/roslyn/pull/12792/files/8804d294acf864ae117d87aa35f93258de50fae2#r72738674